### PR TITLE
Added functionality to move categories

### LINKED ISSUE #2691: Move categories
| Q | A |
| --- | --- |
| Bug? | no |
| New Feature? | yes |
| Sulu Version | 1.2 |
| Browser Version | ... |
#### Expected Behavior

I want to move categories in the sulu admin. The links to categories should stay after move.
#### Possible Solutions

Add a button to the toolbar.

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #2691
| Related issues/PRs | https://github.com/massiveart/husky/pull/765
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR add the functionality to move categories.

#### TODO

- [x] Styles to css
- [x] Husky datagrid deselect radio
- [x] Translations
  * NR: Naar het hoogste niveau verplaatsen
  * FR: Déplacer au plus haut niveau.

### CODE REVIEW

File: src/Sulu/Bundle/CategoryBundle/Category/CategoryManager.php [hunk 1]
```diff
@@ -444,6 +444,26 @@ public function delete($id)
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function move($id, $parent)
+    {
+        if (!$entity = $this->categoryRepository->findCategoryById($id)) {
```
Comment (danrot) [inline src/Sulu/Bundle/CategoryBundle/Category/CategoryManager.php]: `$category` instead of `$entity`

File: src/Sulu/Bundle/CategoryBundle/Category/CategoryManager.php [hunk 2]
```diff
@@ -444,6 +444,26 @@ public function delete($id)
...
+            throw new CategoryIdNotFoundException($id);
+        }
+
+        $parentEntity = null;
```
Comment (danrot) [inline src/Sulu/Bundle/CategoryBundle/Category/CategoryManager.php]: `$parentCategory`

File: src/Sulu/Bundle/CategoryBundle/Resources/public/js/components/categories/list/main.js [hunk 3]
```diff
@@ -37,7 +41,29 @@ define(function () {
                 toolbar: {
                     buttons: {
                         add: {},
-                        deleteSelected: {},
+                        deleteSelected: {
```
Comment (danrot) [inline src/Sulu/Bundle/CategoryBundle/Resources/public/js/components/categories/list/main.js]: Why is this change necessary now?

File: src/Sulu/Bundle/CategoryBundle/Resources/public/scss/modules/move-overlay.scss [hunk 4]
```diff
@@ -0,0 +1,6 @@
+.move-categories {
+    .move-categories-container {
+        height: calc(100vh - 250px);
```
Comment (danrot) [inline src/Sulu/Bundle/CategoryBundle/Resources/public/scss/modules/move-overlay.scss]: Have we ever used `vh`? Seems like IE10 is not supporting it at all...

File: src/Sulu/Bundle/CategoryBundle/Resources/public/js/components/categories/list/main.js (hunk 3 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/CategoryBundle/Resources/public/js/components/categories/list/main.js]: this is only a replacement of hearing on the `sulu.header.delete` event.

File: src/Sulu/Bundle/CategoryBundle/Resources/public/scss/modules/move-overlay.scss (hunk 4 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/CategoryBundle/Resources/public/scss/modules/move-overlay.scss]: TBH i have no idea why i have added this. the overlay handles the bigger container out of the box. i have removed it

**Comment by wachterjohannes** [comment]
@danrot done

**Comment by danrot** [comment]
I don't know why, but all PHP7 tests are failing. Can you investigate?

File: src/Sulu/Bundle/CategoryBundle/Resources/public/js/components/categories/list/move-overlay/main.js [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        bindCustomEvents: function() {
+            this.sandbox.on('husky.datagrid.' + constants.name + '.item.select', function(id) {
+                this.sandbox.emit('husky.overlay.' + constants.name + '.okbutton.activate');
+
+                this.selectedId = id;
+                this.$rootCheckbox.prop('checked', false);
+            }.bind(this));
+
+            this.sandbox.once('husky.overlay.' + constants.name + '.initialized', function() {
+                var fieldsResponse = $.ajax('/admin/api/categories/fields?locale=' + this.options.locale, {async: false});
```
Comment (danrot) [inline src/Sulu/Bundle/CategoryBundle/Resources/public/js/components/categories/list/move-overlay/main.js]: Since when do we use synchronous requests?
